### ISSUE TITLE
adding new options documentation

### ISSUE THREAD

**Issue by tyronerees** [fitbenchmarking/fitbenchmarking#703]
If you want to add a new 'valid' option to an existing option, the instructions in the documentation doesn't seem to work:
https://fitbenchmarking.readthedocs.io/en/latest/extending/options_extend.html

In particular, amending the options class to include
```
VALID_JACOBIAN = \
        {'scipy': ['2-point', '3-point', 'cs'],
         'analytic': ['cutest'],
         'numdifftools': ['numdifftools']}
```
and setting the options to be 
```
[JACOBIAN]

analytic: cutest
scipy: 2-point
numdifftools: numdifftools
```
should (I think) be all that's needed, following the docs.  However, this gives the output:
```
Error while running FitBenchmarking. Exiting. See below for more information.
Failed to process options.
Details: Invalid options key set:
['analytic', 'scipy', 'numdifftools'],
 the valid keys are:
['scipy', 'analytic']
```

**Comment by AndrewLister-STFC** [comment]
It looks like the defaults also weigh in on deciding whats a valid key

**Comment by tyronerees** [comment]
I've got this working now -- it seems to be an issue with the Jacobian options only.  I think we just need a bit more detail in the adding Jacobians docs -- I'll fix this alongside the new docs for fixing #571
